apps/video/js/video.js

```diff
@@ -114,6 +115,17 @@ document.addEventListener('visibilitychange', function visibilityChange() {
     }
   }
   else {
+    // Bug 1151775
+    // If the app is coming to the foreground before a sharing activity is
+    // complete, restore the video (the video will have been released before
+    // initiating the sharing activity and is only restored when the sharing
+    // activity is completed). When the sharing activity is complete and
+    // restoreVideo is again called, the restoreVideo function takes care of
+    // not restoring the video when it doesn't need to be restored.
+    if (sharingActivity) {
+      videoHardwareReleased = true; // Otherwise restoreVideo will abort
+      restoreVideo();
+    }
```

Comment:
I don't understand why you need to make sharingActivity global here.  Isn't videoHardwareReleased the only state variable you need? That can you just do this here:

```
if (videoHardwareReleased) {
  restoreVideo();
}
```
Actually, I think the code you've currently got here is not right. Remember that disposition=window activities like the email app may never complete.  So once you have set sharingActivity, it may never get set back to null.  And in that case, every time the app goes to the background and comes back to the foreground, you are going to force restoreVideo to run. Even if videoHardwareReleased is false, you're forcing restoreVideo to run, which is presumably going to have a big performance impact on how long it takes the user to get back into the video they are watching.

The idea of having a single state variable to hold the current share activity doesn't make sense when the activities might be disposition=window because you can actually have any number of activities running at once.